sklearn/semi_supervised/_self_training.py

```diff
@@ -402,16 +510,65 @@ def score(self, X, y):
         y : array-like of shape (n_samples,)
             Array representing the labels.
 
+        **params : dict of str -> object
+            Parameters requested and accepted by base estimator's score function.
+
+            .. versionadded:: 1.5
+                Only available if `enable_metadata_routing=True`,
+                which can be set by using
+                ``sklearn.set_config(enable_metadata_routing=True)``.
+                See :ref:`Metadata Routing User Guide <metadata_routing>` for
+                more details.
+
         Returns
         -------
         score : float
             Result of calling score on the `base_estimator`.
         """
         check_is_fitted(self)
+        _raise_for_params(params, self, "score")
+
+        if _routing_enabled():
+            # metadata routing is enabled.
+            routed_params = process_routing(self, "score", **params)
+        else:
+            routed_params = Bunch()
+            routed_params.base_estimator = Bunch(score={})
+
         X = self._validate_data(
             X,
             accept_sparse=True,
             force_all_finite=False,
             reset=False,
         )
-        return self.base_estimator_.score(X, y)
+        return self.base_estimator_.score(X, y, **routed_params.base_estimator["score"])
+
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.5
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = MetadataRouter(owner=self.__class__.__name__)
+        router.add(
+            base_estimator=self.base_estimator,
+            method_mapping=(
+                MethodMapping()
+                .add(callee="fit", caller="fit")
+                .add(callee="fit", caller="score")
```

Comment:
I think if you switch `caller` and `callee` everywhere, then this is correct. As in, `fit` calls `score`, but `score` doesn't call `fit`.

Reply:
Ah I see. Thanks!

Reply:
Changed here.